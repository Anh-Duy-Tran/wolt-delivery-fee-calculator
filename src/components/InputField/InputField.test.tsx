import { expect, it, describe, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { InputField } from '.';
import React from 'react';

describe('InputField Component', () => {
  it('should render correct to the snapshot', () => {
    const { asFragment } = render(<InputField label="Test Label" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should update value on change', () => {
    render(<InputField label="Test Label" />);
    const input = screen.getByTestId('input-field') as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'New Value' } });
    expect(input.value).toBe('New Value');
  });

  it('should call onChange handler', () => {
    const handleChange = vi.fn();
    render(<InputField label="Test Label" onChange={handleChange} />);
    const input = screen.getByTestId('input-field');
    fireEvent.change(input, { target: { value: 'New Value' } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('should display error message when provided', () => {
    const errorMessage = 'Error message';
    render(<InputField label="Test Label" errorMessage={errorMessage} />);
    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });

  it('should forward additional props to input', () => {
    render(<InputField label="Test Label" placeholder="Enter text" />);
    expect(screen.getByPlaceholderText(/enter text/i)).toBeInTheDocument();
  });

  it('should forward ref to input element', () => {
    const ref = React.createRef<HTMLInputElement>();
    render(<InputField label="Test Label" ref={ref} />);
    expect(ref.current).toBeInstanceOf(HTMLInputElement);
  });
});
