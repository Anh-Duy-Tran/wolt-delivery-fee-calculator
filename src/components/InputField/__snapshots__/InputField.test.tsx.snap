// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputField Component > renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="relative flex w-full items-center justify-center"
    >
      <input
        class="peer w-full rounded-md border-2 border-wolt-grey p-4 pb-[6px] pt-[22px] outline-none transition placeholder:opacity-0 hover:border-wolt-blue focus:border-wolt-blue focus:shadow-border-like focus:placeholder:opacity-100"
        data-testid="input-field"
      />
      <label
        class="linear duration-config pointer-events-none absolute w-full origin-left translate-x-4 scale-100 
        select-none opacity-60 transition-all peer-focus:-translate-y-3 peer-focus:scale-[65%] false"
      >
        Test Label
      </label>
    </div>
  </div>
</DocumentFragment>
`;
